.github/workflows/generate-crd.yml

```diff
@@ -11,40 +14,103 @@ jobs:
       id-token: write
       contents: read
     steps:
+      - name: Determine operator tag
+        id: set_operator_tag
+        run: |
+          if [ -n "${{ github.event.inputs.tag }}" ]; then
+            TAG="${{ github.event.inputs.tag }}"
+          else
+            TAG="${{ github.event.release.tag_name }}"
+          fi
+          echo "Operator tag: $TAG"
+          echo "OPERATOR_TAG=$TAG" >> $GITHUB_ENV
```

Comment:
Ah this is going to break for soon for ~3 different reasons:
1. The tags we will be using going forward will be prefixed with `operator/` (e.g. `operator/v2.4.3`)
2. The literal tags and the external facing versions will be different post 2.4.x. `operator/v25.1.2` will be the tag but the image and "version" of the operator will technically be `v25.1-k8s2`
3. We're going to be tagging non-operator releases as well 😓

I'm not sure the best way to make a workflow in GHA short circuit but you'll want to filter on tags having the prefix `operator/`.

Oh, looks like 2 won't matter as you're just using the tag to pull the operator repo! Double check me though.